strawberry/types/fields/resolver.py

```diff
@@ -344,6 +345,10 @@ def is_graphql_generic(self) -> bool:
 
     @cached_property
     def is_async(self) -> bool:
+        if hasattr(inspect, "markcoroutinefunction"):
+            iscoroutinefunction = inspect.iscoroutinefunction
+        else:
+            iscoroutinefunction = asyncio.iscoroutinefunction
```

Comment:
nit: I would define this at module level

Also, maybe add a comment to why this is being done. I know why, but someone checking this `hasattr(inspect, "markcoroutinefunction")` to define `iscoroutinefunction` could get confused